rest_sensors.cpp

```diff
@@ -1089,12 +1089,14 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
                         // For newer models it is not possible to write to this attribute.
                         // Newer models must use the standard Occupied Heating Setpoint value (0x0012) using a default (or none) manufacturer.
                         // See GitHub issue #1098
-                        bool success = sensor->swVersion().toInt() < 22190930 ?
-                            addTaskThermostatReadWriteAttribute(task, deCONZ::ZclWriteAttributesId, VENDOR_JENNIC, 0x4003, deCONZ::Zcl16BitInt, heatsetpoint) :
-                            addTaskThermostatReadWriteAttribute(task, deCONZ::ZclWriteAttributesId, VENDOR_NONE, 0x0012, deCONZ::Zcl16BitInt, heatsetpoint);
-
-                        if (success)
+                        // UPD 16-11-2020: Since there is no way to reckognize older and newer models correctly and a new firmware version is on its way this
+                        //                 'fix' is changed to a more robust but ugly implementation by simply sending both codes to the device. One of the commands
+                        //                 will be accepted while the other one will be refused. Let's hope this code can be removed in a future release.
+    
+                        if (addTaskThermostatReadWriteAttribute(task, deCONZ::ZclWriteAttributesId, VENDOR_JENNIC, 0x4003, deCONZ::Zcl16BitInt, heatsetpoint) &&
```

Comment:
I'm not sure whether a logical AND is correct here? I think the writing to 0x4003 will fail on those new devices with wrong firmware and therefor this statement want get true?

Reply:
The method `addTaskThermostatReadWriteAttribute` returns `true` whenever a correct request message has been send. It does not depend on the response of the client. So, as far as I know, as long as the request message is syntactically correct the method will return `true`. Since this message is constructed in the code (apart from the request heating setpoint) I think it will always succeed (unless you enter out-of-range heating setpoint values).

Reply:
In other words: having a logical `or` here would only evaluate the left hand side (since it most likely returns `true`) and ignore the right hand side discarding one of the two requests.

Reply:
Thanks @joukestoel for pointing this out for me. I'm really new with the deconz code. And have no running Dev-environment yet, because I mainly programming in Java and Python. 

You can set my comment to resolved.

Reply:
👍